### PR TITLE
Disable jaxrs annotation instrumentation when controller telemetry is not enabled

### PR DESCRIPTION
**Pull request by laurit**

### CODE REVIEW

File: instrumentation/jaxrs/jaxrs-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxrsInstrumentationModule.java [hunk 1]
```diff
@@ -31,4 +33,9 @@ public ElementMatcher.Junction<ClassLoader> classLoaderMatcher() {
   public List<TypeInstrumentation> typeInstrumentations() {
     return singletonList(new JaxrsAnnotationsInstrumentation());
   }
+
+  @Override
+  public boolean defaultEnabled(ConfigProperties config) {
+    return ExperimentalConfig.get().controllerTelemetryEnabled();
```
Comment (trask) [inline instrumentation/jaxrs/jaxrs-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxrsInstrumentationModule.java:39]: I liked the comment you added in the other PRs "this instrumentation only produces controller telemetry"

**Review by trask — APPROVED** [review]

File: instrumentation/jaxrs/jaxrs-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxrsInstrumentationModule.java (hunk 1 above)
Comment (trask) [reply instrumentation/jaxrs/jaxrs-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxrsInstrumentationModule.java:39]: it looks like this instrumentation also sets `http.route`?

Comment (laurit) [reply instrumentation/jaxrs/jaxrs-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxrsInstrumentationModule.java:39]: > I liked the comment you added in the other PRs "this instrumentation only produces controller telemetry"

As you noted that comment wouldn't be strictly true here as this instrumentation does a bit more.

> it looks like this instrumentation also sets http.route?

It does set it providing it is not already set. As we have instrumentations for jersey/resteasy etc. there is a good chance that the route has already been set by the time this code runs. This instrumentation could be useful is there was a jax-rs framework that we don't have and instrumentation for.

Comment (trask) [reply instrumentation/jaxrs/jaxrs-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxrsInstrumentationModule.java:39]: ah, I see, can you add a comment about that? (and also mention I assume that the upside to disabling it is to improve startup overhead?)

Comment (laurit) [reply instrumentation/jaxrs/jaxrs-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v1_0/JaxrsInstrumentationModule.java:42]: comment added

**Review by trask — APPROVED** [review]

**Pull request merged**